src/DefinitionResolver.php

```diff
@@ -33,35 +41,43 @@ public function __construct(ReadableIndex $index)
     {
         $this->index = $index;
         $this->typeResolver = new TypeResolver;
-        $this->prettyPrinter = new PrettyPrinter;
+        $this->docBlockFactory = DocBlockFactory::createInstance();
     }
 
     /**
-     * Builds the declaration line for a given node
+     * Builds the declaration line for a given node. Declarations with multiple lines are trimmed.
      *
      * @param Node $node
      * @return string
      */
-    public function getDeclarationLineFromNode(Node $node): string
+    public function getDeclarationLineFromNode($node): string
     {
-        if ($node instanceof Node\Stmt\PropertyProperty || $node instanceof Node\Const_) {
-            // Properties and constants can have multiple declarations
-            // Use the parent node (that includes the modifiers), but only render the requested declaration
-            $child = $node;
-            /** @var Node */
-            $node = $node->getAttribute('parentNode');
-            $defLine = clone $node;
-            $defLine->props = [$child];
+        // If node is part of a declaration list, build a declaration line that discludes other elements in the list
+        //  - [PropertyDeclaration] // public $a, [$b = 3], $c; => public $b = 3;
+        //  - [ConstDeclaration | ClassConstDeclaration] // "const A = 3, [B = 4];" => "const B = 4;"
+        if (
+            ($declaration = ParserHelpers::tryGetPropertyDeclaration($node)) && ($elements = $declaration->propertyElements) ||
+            ($declaration = ParserHelpers::tryGetConstOrClassConstDeclaration($node)) && ($elements = $declaration->constElements)
+        ) {
+            $defLine = $declaration->getText();
+            $defLineStart = $declaration->getStart();
+
+            $defLine = \substr_replace(
+                $defLine,
+                $node->getFullText(),
+                $elements->getFullStart() - $defLineStart,
+                $elements->getFullWidth()
+            );
         } else {
-            $defLine = clone $node;
-        }
-        // Don't include the docblock in the declaration string
-        $defLine->setAttribute('comments', []);
-        if (isset($defLine->stmts)) {
-            $defLine->stmts = [];
+            $defLine = $node->getText();
         }
-        $defText = $this->prettyPrinter->prettyPrint([$defLine]);
-        return strstr($defText, "\n", true) ?: $defText;
+
+        // Trim string to only include first line
+        $defLine = \rtrim(\strtok($defLine, "\n"), "\r");
```

Comment:
Leading/trailing whitespace is already gone by this point, this is handling a specific case.